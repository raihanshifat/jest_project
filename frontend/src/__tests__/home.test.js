import {cleanup, fireEvent, render,screen } from '@testing-library/react';
import Home from '../components/home'
import renderer from 'react-test-renderer'
import {BrowserRouter as Router} from 'react-router-dom'
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import React from 'react';

Enzyme.configure({ adapter: new Adapter() });
beforeEach(()=>cleanup())

it("Home page snapshot testing when user is unauthorized",()=>{
    const tree=renderer.create(<Router><Home></Home></Router>).toJSON()
    expect(tree).toMatchSnapshot();
    
})

it("Home page testing when user authorized",()=>{
    React.useState=jest.fn().mockImplementation(()=>[[{id:1}],jest.fn()])
    render(<Router><Home></Home></Router>);
    expect(screen.getByText("logout")).toBeInTheDocument();
    
})