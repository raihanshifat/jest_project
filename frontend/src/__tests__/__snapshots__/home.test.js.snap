// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page snapshot testing when user is unauthorized 1`] = `
<div>
  <h1>
    Please login to view the homepage
  </h1>
  <a
    aria-current={null}
    href="/signup"
    onClick={[Function]}
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        SignUp
      </span>
    </button>
  </a>
  <a
    aria-current={null}
    href="/login"
    onClick={[Function]}
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        SignIn
      </span>
    </button>
  </a>
</div>
`;
